Fix seed command unrecognized "number" argument

diogoosorio opened the pull request:
As mentioned by @Brobin on the issue #37 , the problem is that [optparse was discontinued in favor of argparse](https://docs.djangoproject.com/en/1.10/releases/1.8/#extending-management-command-arguments-through-command-option-list) when transitioning from Django 1.7 to 1.8.

The 2 flavors were supported until the version 1.10 was released. Here's the [commit which discontinued the support of optarse](https://github.com/django/django/commit/6a70cb53971a19f2d9e71d5ee24bfb0e844b4d9d#diff-dfc45ab8548a0777543d12d6e77c9173L230).

This is what happening "under the hood" (as far as I can tell):

* Django <= 1.7 - optparse is used, the property `option_list` is consumed

* Django 1.8 - 1.9 - if `option_list` is present optparse is used and the method `add_arguments` is never invoked

* Django 1.10 - the support for `option_list` is removed completely

I'm not completely happy with the change I'm proposing here (having the argument definition duplicated is rather dirty), but it works.

Ideally I'd argue that releasing a new major of the library assuming that Django 1.7 is no longer supported ([as that version of the framework itself is no longer maintained](https://www.djangoproject.com/download/#supported-versions)) would be the best solution to avoid poluting the codebase with [old version-based conditional statements](https://github.com/Brobin/django-seed/blob/master/django_seed/management/commands/seed.py#L45).

If 1.7 support is to be maintained, I'm open to suggestions on a cleaner approach to address the issue. 

PS - Tests are passing under Django 1.10 (that's why I've added them to the CI recipe) but a warning is being emitted.

Brobin commented:
I'm all for dropping 1.7 support, if you are willing to make that change. Make sure to also remove 1.7 from the build list when updating it.

diogoosorio commented:
Ok, I'll give it a go during the weekend - if you don't mind leaving the PR open until then, I'll add the changes to this branch for you to review.

diogoosorio commented:
Sorry for the late reply (lots of work, lack of time, yada, yada)...

Anyway feel free to preform the code review whenever you like. I went a bit out of scope but I was seeing warnings regarding the timezone "awareness" when seeding the database (both when running the tests and when executing the CLI task in my application).

Here's what I ended up doing:

* Cleaned up all the references/conditional statements regarding Django <= 1.7 (this was what was previously discussed)

* Updated the CI recipe/setup.py to reflect the change

* Updated the dependency used to generate the actual fake data. [fake-factory ](https://pypi.python.org/pypi/fake-factory) is marked as being deprecated and points to the [Faker](https://pypi.python.org/pypi/Faker/0.7.7) package as an alternative. The API of the library doesn't appear to have changed (and if did, hopefully I'd see some tests failing).

* [This test](https://github.com/Brobin/django-seed/blob/0304a1d40869d0811098104bf2c799e5b1b2c9fe/django_seed/tests.py#L111) should be failing but it isn't. The behavior is [documented here](https://docs.djangoproject.com/en/1.10/topics/i18n/timezones/#interpretation-of-naive-datetime-objects) - if the USE_TZ setting is set to true the framework will attempt to localize the date when the object is persisted (thus the test was always yielding true, even if the code was generating a "naive" datetime instance).

* The approach to solve the previous problem was to decouple the "guesser" tests (which are responsible for generating the timezone-aware data) from the the "seeder" tests.

Brobin commented:
Thanks for the thorough testing and everything on this!
